src/Microsoft.Data.Analysis/StringDataFrameColumn.cs

```diff
@@ -467,5 +467,30 @@ protected internal override Delegate GetDataViewGetter(DataViewRowCursor cursor)
 
         private ValueGetter<ReadOnlyMemory<char>> CreateValueGetterDelegate(DataViewRowCursor cursor) =>
             (ref ReadOnlyMemory<char> value) => value = this[cursor.Position].AsMemory();
+
+        private ValueGetter<ReadOnlyMemory<char>> getter = null;
```

Comment:
This shouldn't be cached on the instance. If you call `ToDataFrame` twice, it isn't going to work.

Reply:
Maybe a better approach would be to have the column return a delegate that the uber IDataView extension holds throughout the call to `ToDataFrame`.... ?

---
In reply to: [593504550](https://github.com/dotnet/machinelearning/pull/5712#discussion_r593504550) [](ancestors = 593504550)

Reply:
Actually I didn't realize that I'd already implemented this. `RowCursor` already holds a list of `_getters` internally. The only wrinkle is that `_getters` is a `List<Delegate>` => in each column we'd have to cast to the delete of the appropriate type before invoking the delegate.